AndroidSDKTests/src/test/java/com/leanplum/internal/RequestOldTest.java

```diff
@@ -371,6 +372,34 @@ public void testGetRequestsWithEncodedStringStoredRequests() {
     return requests;
   }
 
+  @Test
+  public void testNotSendingIfContextIsNull() {
+    Context context = Leanplum.getContext();
+    Leanplum.setApplicationContext(null);
+
+    final Semaphore semaphore = new Semaphore(1);
+    semaphore.tryAcquire();
+
+    // Given a request.
+    Map<String, Object> params = new HashMap<>();
+    params.put("data1", "value1");
+    params.put("data2", "value2");
+    RequestOld request = new RequestOld(POST, Constants.Methods.START, params);
+    request.onError(new RequestOld.ErrorCallback() {
```

Comment:
Will this get called in this instance? If we're returning early in `saveRequestForLater`, then you won't be throwing an error, right?

Reply:
it will throws error first then tries to save for later